Explicitly set entity name for VenstarSensor

dseven opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

Address:

[homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #100295
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

dseven committed:
VenstarSensor entity name

Set entity name using _attr_name instead of a name property, to
avoid warnings about name being implicitly set to None.

rainepretorius reviewed (approved):
Looks all good. WHy do you want to update the _attr_name to None? It will cause multiple names on the device name?

joostlek reviewed (changes requested):
This is not the right way to solve this issue. This will rename every sensor to the device name, causing `sensor.{device_name}` and `sensor.{device_name}_1`. This means one of the entities created doesn't have a translation key or device class (or name)

joostlek commented:
I just checked the code, and every sensor does have a name, so I am curious where this message comes from

joostlek commented:
So to explain it more, this message only pops up when the entity has `has_entity_name = True` and has no name. translation key or device class (in this specific order). 

```python
    @property
    def name(self):
        """Return the name of the device."""
        return self.entity_description.name_fn(self.sensor_name)
```
This part of the code makes sure the code always returns a name. This function never seems to return `None`.

I am guessing you can set this integration up in a development environment. Can you maybe give it a shot to change that piece of code to this?

```python
    @property
    def name(self):
        """Return the name of the device."""
        name = self.entity_description.name_fn(self.sensor_name)
        print(name)
        return name
```
This would print out the name of the sensor before being returned. If I have to believe the code, this should never print `None` in the logs. If it does, we have something to investigate.

dseven commented:
FWIW, the proposed change did make the warning go away for me, did not generate any new errors, and didn't appear to affect functionality at all.

I changed the diagnostic code to use logging instead of print() (and reverted my change). Here's what I get on startup...

```homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:11:18.344 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 1 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 1 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 2 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 2 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 1 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 1 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 2 Runtime
homeassistant  | 2023-10-17 07:11:18.345 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22
homeassistant  | 2023-10-17 07:11:18.346 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 2 Runtime
homeassistant  | 2023-10-17 07:11:18.346 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Schedule Part
homeassistant  | 2023-10-17 07:11:18.346 WARNING (MainThread) [homeassistant.helpers.entity] Entity None (<class 'homeassistant.components.venstar.sensor.VenstarSensor'>) is implicitly using device name by not setting its name. Instead, the name should be set to None, please create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+venstar%22
homeassistant  | 2023-10-17 07:11:18.346 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Schedule Part
homeassistant  | 2023-10-17 07:12:20.972 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:12:20.972 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:12:20.973 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 1 Runtime
homeassistant  | 2023-10-17 07:12:20.973 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 2 Runtime
homeassistant  | 2023-10-17 07:12:20.973 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 1 Runtime
homeassistant  | 2023-10-17 07:12:20.973 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 2 Runtime
homeassistant  | 2023-10-17 07:12:20.973 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Schedule Part
homeassistant  | 2023-10-17 07:13:24.184 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:13:24.184 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:13:24.184 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 1 Runtime
homeassistant  | 2023-10-17 07:13:24.184 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 2 Runtime
homeassistant  | 2023-10-17 07:13:24.184 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 1 Runtime
homeassistant  | 2023-10-17 07:13:24.185 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 2 Runtime
homeassistant  | 2023-10-17 07:13:24.185 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Schedule Part
homeassistant  | 2023-10-17 07:14:27.327 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Thermostat Temperature
homeassistant  | 2023-10-17 07:14:27.327 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Space Temperature
homeassistant  | 2023-10-17 07:14:27.327 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 1 Runtime
homeassistant  | 2023-10-17 07:14:27.327 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Heating Stage 2 Runtime
homeassistant  | 2023-10-17 07:14:27.327 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 1 Runtime
homeassistant  | 2023-10-17 07:14:27.328 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Cooling Stage 2 Runtime
homeassistant  | 2023-10-17 07:14:27.328 WARNING (MainThread) [homeassistant.components.venstar.const] My name is Schedule Part```

joostlek commented:
Heh, this doesn't make sense

joostlek commented:
Right, I looked it up and I might found something. `has_entity_name` doesn't really like the `name` property.

Can you try `self._attr_name = entity_description.name_fn(sensor_name)` in the constructor of the entity and remove the name property?

dseven commented:
> Can you try `self._attr_name = entity_description.name_fn(sensor_name)` in the constructor of the entity and remove the name property?

That seems to make everything happy. If I don't hear any counter arguments, I'l update the commit in a bit.

joostlek commented:
I copied this from Unifi, so I would say go for it

joostlek reviewed (approved):
If you put this ready for review I will merge this
